From 5efe798f4bb3ab403183f4d4dc80357cb1f34525 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Wed, 18 Nov 2020 14:19:29 -0800
Subject: [PATCH 22/31] Avoid calling setting target for invalid participant

When participant name doesn't get a valid cdev, don't call
to set_adaptive_target as it will segfault.

Signed-off-by: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
(cherry picked from commit f4ab0673bf68eec9f041e3988bef6e0bf8c06518)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 src/thd_engine_adaptive.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/thd_engine_adaptive.cpp b/src/thd_engine_adaptive.cpp
index b7709e7..2be02dc 100644
--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -1390,7 +1390,8 @@ void cthd_engine_adaptive::execute_target(struct adaptive_target target) {
 		return;
 	}
 	thd_log_info("target:%s %d\n", target.code.c_str(), argument);
-	cdev->set_adaptive_target(target);
+	if (cdev)
+		cdev->set_adaptive_target(target);
 }
 
 void cthd_engine_adaptive::exec_fallback_target(int target){
-- 
2.27.0

